src/org/opensolaris/opengrok/analysis/plain/PlainAnalyzer.java

```diff
@@ -134,4 +140,51 @@ public void analyze(Document doc, StreamSource src, Writer xrefOut)
             }
         }
     }
+
+    private void tryAddingDefs(Document doc, Definitions defs, StreamSource src,
+        String fullpath) throws IOException {
+
+        DefinitionsTokenStream defstream = new DefinitionsTokenStream();
+        defstream.initialize(defs, src, (reader) -> wrapReader(reader));
+
+        /**
+         *     Testing showed that UnifiedHighlighter will fall back to
+         * ANALYSIS in the presence of multi-term queries (MTQs) such as
+         * prefixes and wildcards even for fields that are analyzed with
+         * POSTINGS -- i.e. with DOCS_AND_FREQS_AND_POSITIONS_AND_OFFSETS.
+         * This is despite UnifiedHighlighter seeming to indicate that
+         * postings should be sufficient in the comment for
+         * shouldHandleMultiTermQuery(String): "MTQ highlighting can be
+         * expensive, particularly when using offsets in postings."
+         *     DEFS re-analysis will not be correct, however, as the
+         * PlainAnalyzer which UnifiedHighlighter will use on-the-fly will
+         * not correctly integrate ctags Definitions.
+         *     Storing term vectors, however, allows UnifiedHighlighter to
+         * avoid re-analysis at the cost of a larger index. As DEFS are a
+         * small subset of source text, it seems worth the cost to get
+         * accurate highlighting for DEFS MTQs.
+         */
+        doc.add(new OGKTextVecField(QueryBuilder.DEFS, defstream));
+    }
+
+    /**
+     * Identical to {@link #getReader(java.io.InputStream)} but overlaying an
+     * existing stream.
+     * @see #getReader(java.io.InputStream)
+     */
+    private Reader wrapReader(Reader reader) {
+        return ExpandTabsReader.wrap(reader, project);
+    }
+
+    private static boolean bytesEqual(byte[] a, byte[] b) {
```

Comment:
Maybe unused? :) If it is used then you can consider using `Arrays.equals()`.

Reply:
ok removed